logging/logging.go

```diff
@@ -2,76 +2,84 @@
 package logging
```

Comment:
[q] Can you describe what's in `logging/logger.go` and what's in `logging/logging.go` now?

Reply:
Sure, sorry for not doing that. Perfectly reasonable for that explanation to have accompanied this PR. Things are still in flux as I settle on code organization. But where I stand now:
* logger.go
  * contains the `Logger` interface
  * contains `ZapCompatibleLogger` interface
  * contains `zLogger` wrapper for ZapCompatible->Logger upconvert
* logging.go
  * Contains the public "static" methods for creating loggers/configurations
 * level.go you didn't ask about
   * But it contains the log Level stuff. Mostly just serialization
 * impl.go does not exist yet in this PR
   * but that's where I'm putting the new Logger implementation that does the things we want (like the upcoming `SetLevel`)
   * I know its customary to make an `impl` directory. If `impl.go` gets large/tangly enough to warrant extra files, I'll go towards a separate directory.

Reply:
That all sounds great; thanks!

> I know its customary to make an impl directory.

I haven't heard of this standard in golang (I've always referred to [`golang-standards/project-layout`](https://github.com/golang-standards/project-layout) for "non-official but good" recommendations on project layouts, and they do not use `impl` directories from what I can tell). In any case, `impl` directory and/or `impl.go` seem reasonable.